common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/IDToken.java

```diff
@@ -219,27 +217,29 @@ public String getRawIDToken() {
         return Collections.unmodifiableMap(mTokenClaims);
     }
 
-    private Map<String, String> parseJWT(final String idToken) throws ServiceException {
-        final String idTokenBody = extractJWTBody(idToken);
-        final byte[] data = Base64.decode(idTokenBody, Base64.URL_SAFE);
-
+    public static Map<String, String> parseJWT(final String idToken) throws ServiceException {
+        final JWTClaimsSet claimsSet;
         try {
-            final String decodedBody = new String(data, Charset.forName(StringExtensions.ENCODING_UTF8));
-            return JsonExtensions.extractJsonObjectIntoMap(decodedBody);
-        } catch (final JSONException e) {
-            throw new ServiceException("", ErrorStrings.INVALID_JWT, e);
+            // Create a SignedJWT from the input token String
+            final SignedJWT signedJWT = SignedJWT.parse(idToken);
```

Comment:
@kreedula This should follow this guidance to support v1's unsigned jwts: https://connect2id.com/products/nimbus-jose-jwt/examples/combined-jwt-parsing